core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/AttributeComponent.scala

```diff
@@ -325,16 +325,31 @@ trait AttributeComponent {
       qb.sqlBuilder += " collate utf8_bin"
     }
 
-    def findUniqueStringsByNameQuery(attrName: AttributeName, queryString: Option[String], limit: Option[Int] = None) = {
-
-      val basicFilter = filter(rec =>
+    /**
+      * Returns all unique attribute values, with the specfied filters applied
+      *
+      * @param attrName the name of the attribute to filter for (note that this is a (namespace, name) pair)
+      * @param queryString the string value to filter for, optional
+      * @param limit the maximum number of results to return, optional
+      * @param ownerIds the ownerIds (i.e. the entity ID or workspace ID that the attribute belongs to) to filter for, optional
+      * @return the attribute values that meet the specified filters
+      */
+    def findUniqueStringsByNameQuery(attrName: AttributeName, queryString: Option[String], limit: Option[Int] = None, ownerIds: Option[Seq[OWNER_ID]] = None) = {
+
+      val basicFilter = filter { rec =>
         rec.namespace === attrName.namespace &&
           rec.name === attrName.name &&
-          rec.valueString.isDefined)
+          rec.valueString.isDefined
+      }
+
+      val withOptionalOwnerFilter =
+        basicFilter.filterOpt(ownerIds) { case (table, ownerIds) =>
+          table.ownerId inSetBind ownerIds
```

Comment:
I need to double check this: potentially having >1000 UUIDs in an `in` clause might be an issue.

Batching might be the more scalable solution here.

Reply:
OK yeah this definitely needs to batch

Reply:
`listWorkspaces` does the same thing without batching - many many ids inside the `in` clause. Are you sure you need to batch - and include all the headache of combining the batched results back into a useful result?

Reply:
I actually just reached this same conclusion myself which is why I blew away the commit that had it partially implemented and resolved this! I tested with 2345 workspaceIds and it was totally fine

Reply:
The reason I thought I needed to batch was because I saw this comment in another area that batches https://github.com/broadinstitute/rawls/blob/f536962c85911d50ae23246d3fd422608d6804e9/core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala#L935

Reply:
oooh, fair point - though it looks like "The number of values in the IN() list is only limited by the [max_allowed_packet](https://dev.mysql.com/doc/refman/5.6/en/server-system-variables.html#sysvar_max_allowed_packet) value." according to https://dev.mysql.com/doc/refman/5.6/en/comparison-operators.html, which means we'd need ~4MB of workspaceIds to hit that limit